include/fastrtps/types/v1_3/MemberDescriptor.h

```diff
@@ -0,0 +1,336 @@
+     * @return standard @ref ReturnCode_t
+     */
+    RTPS_DllAPI ReturnCode_t copy_from(
+            const MemberDescriptor& other);
+
+    /**
+     * checks for equality according with [standard] section 7.5.2.7.4
+     * @param[in] other @ref MemberDescriptor reference to compare to
+     * @remark the @ref MemberDescriptor::equals relies on this method
+     * @return bool on equality
+     * [standard]: https://www.omg.org/spec/dds-xtypes/1.3/ "OMG standard"
+     */
+    RTPS_DllAPI bool operator ==(
+            const MemberDescriptor& other) const;
+
+    /**
+     * checks for inequality according with [standard] section 7.5.2.7.4
+     * @param[in] other @ref MemberDescriptor reference to compare to
+     * @remark the @ref MemberDescriptor::equals relies on this method
+     * @return bool on inequality
+     * [standard]: https://www.omg.org/spec/dds-xtypes/1.3/ "omg standard"
+     */
+    RTPS_DllAPI bool operator !=(
+            const MemberDescriptor& other) const;
+
+    /**
+     * State comparison
+     * @remarks using `==` and `!=` operators is more convenient
+     * @param[in] other @ref MemberDescriptor object whose state to compare to
+     * @return \b bool `true` on equality
+     * [standard]: https://www.omg.org/spec/dds-xtypes/1.3/ "omg standard"
+     */
+    RTPS_DllAPI bool equals(
+            const MemberDescriptor& other) const;
+
+    //! convenient getter for the associated \b type property
+    RTPS_DllAPI TypeKind get_kind() const;
+
+    /**
+     * Queries the desired or actual member id
+     * @return MemberId id
+     */
+    RTPS_DllAPI MemberId get_id() const;
+
+    /**
+     * Queries the desired or actual member position in the collection.
+     * @return uint32_t position
+     */
+    RTPS_DllAPI uint32_t get_index() const;
+
+    /**
+     * Queries the desired or actual member name
+     * @return std::string name
+     */
+    RTPS_DllAPI std::string get_name() const;
+
+    //! getter for the labels member
+    RTPS_DllAPI const std::set<uint64_t>& get_union_labels() const;
+
+    //! getter for the \b default_value property
+    RTPS_DllAPI std::string get_default_value() const;
+
+    //! checks if the member is the 'default' idl case
+    RTPS_DllAPI bool is_default_union_value() const;
+
+    /**
+     * Tests state consistency
+     * @remark A MemberDescriptor shall be considered consistent if and only if all of the values
+     *         of its properties are considered consistent with its collection owner
+     * @param[in] parentKind @ref eprosima::fastrtps::types::TypeKind collection's owner kind
+     * @return bool `true` if consistent
+     */
+    RTPS_DllAPI bool is_consistent(
+            TypeKind parentKind) const;
+
+    //! insert a new label for this union member
+    RTPS_DllAPI void add_union_case_index(
+            uint64_t value);
+
+    /**
+     * Set member @ref MemberId
+     * @param[in] id desired MemberId
+     */
+    RTPS_DllAPI void set_id(
```

Comment:
I think is easier to user to have getter and setter together in the header file.